src/lib/problem_report.c

```diff
@@ -1241,7 +1263,7 @@ problem_formatter_generate_report(const problem_formatter_t *self, problem_data_
                     self->pf_default_summary);
 
         format_percented_string(self->pf_default_summary,
-                   data, problem_report_get_buffer(pr, PR_SEC_SUMMARY));
+                   data, problem_report_get_buffer(pr, PR_SEC_SUMMARY), self->fmt_file);
```

Comment:
I had an idea for this: what if we started using `GError`s more liberally and just propagated to callers? They will surely all have a pointer to the path somewhere. The downside is that you will need to add error handling to all of them, but a cursory glance tells me that it might already be done.

Reply:
https://developer.gnome.org/glib/stable/glib-Error-Reporting.html for documentation.

Reply:
Though another concern is breaking ABI compatibility, but, in Fedora at least, no one seems to be using libreport meaningfully (aside from us).